props/blaster.h

```diff
@@ -260,9 +266,39 @@ class Blaster : public PROP_INHERIT_PREFIX PropBase {
     }
   }
 
+  RefPtr<BufferedWavPlayer> tmp;
+  uint32_t len;
+
   void Loop() override {
     PropBase::Loop();
     PollNextAction();
+	  
+    STATE_MACHINE_BEGIN();
+    while (true) {
+      if (auto_firing_) {
+        tmp = GetWavPlayerPlaying(&SFX_auto);
```

Comment:
calling this every loop is a bit expensive.
it would be better if we got the length of the wav file right after we start playing the sound.

Reply:
Doesn't that happen elsewhere though? All that's here is DoBeginLockup() and SFX_auto isn't playing yet.

Reply:
Well, we can keep the length, or the player, basically something like:

if (auto_firing_) {
  if (!auto_player_) {
     auto_player_ = GetWavPlayerPlaying(&SFX_auto);
  }
  // rest of logic here
} else {
   auto_player_.reset();
}

Reply:
ah. I had `if (auto_firing_ && GetWavPlayerPlaying(&SFX_auto)) {
`
I will try this, thanks.

Reply:
Actually, that avoids NULL POINTER while bgnauto plays.
Well, latest update works, but does it make sense?